components/lib/dataprotect/src/main/java/mozilla/components/lib/dataprotect/KeySharedPreferences.kt

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.dataprotect
+
+import android.annotation.TargetApi
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import android.os.Build
+import android.os.Build.VERSION_CODES.M
+import android.util.Base64
+import mozilla.components.support.base.log.logger.Logger
+import java.nio.charset.StandardCharsets
+import java.security.GeneralSecurityException
+
+private const val BASE_64_FLAGS = Base64.URL_SAFE or Base64.NO_PADDING
+
+/**
+ * A wrapper around [SharedPreferences] which encrypts contents on supported API versions (23+).
+ * Otherwise, this simply delegates to [SharedPreferences].
+ *
+ * @param context Application context, used for accessing [SharedPreferences].
```

Comment:
Can we reword this slightly because getApplicationContext is a thing in Android and this could be confusing